### PR TITLE
Feature/categories2

### PR DESCRIPTION
**Pull request by tepoorte**
categories

### CODE REVIEW

File: app/src/main/java/com/studhub/app/data/repository/CategoryRepositoryImpl.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    private val categories: List<Category> = listOf(
+        Category(id = 1, name = "electronics", "phones/cameras/.. "),
+        Category(id = 2, name = "school items","books/pencils/bags/.."),
+        Category(id = 3, name = "accessories","keys/necklaces/.."),
+        Category(id = 4, name = "insruments","ear phones/guitar/.."),
+        Category(id = 5, name = "mobility","bikes/scooter/.."),
+        Category(id = 6, name = "clothes","pants/shirts/.."),
+        Category(id = 7, name = "artdecorations","paintings/tapis/.."),
+        Category(id = 8, name = "services","online services/apps/supports/.."),
+        Category(id = 9, name = "other","other")
```
Comment (edouardmichelin) [inline app/src/main/java/com/studhub/app/data/repository/CategoryRepositoryImpl.kt:19]: Add a space after commas - like the first line.
Also add parameter name for `description`, e.g.: `Category(id = 3, name = "accessories", description = "keys/necklaces/..")`

**Review by edouardmichelin — CHANGES_REQUESTED** [review]
Some minor changes to add.
Otherwise as everything has already been seen with you, lgtm

**Review by Edouard-6 — COMMENTED** [review]
A minor change, otherwise LGTM

File: app/src/main/java/com/studhub/app/data/repository/ListingRepositoryImpl.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        listingId: Long,
+        updatedListing: Listing
+    ): Flow<ApiResponse<Listing>> {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun removeListing(listingId: Long): Flow<ApiResponse<Boolean>> {
+        TODO("Not yet implemented")
+    }
+}
```
Comment (edouardmichelin) [inline app/src/main/java/com/studhub/app/data/repository/ListingRepositoryImpl.kt:31]: Remove file

File: app/src/androidTest/java/com/studhub/app/CategoryRepositoryImplTest.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                }
+            }
+        }
+
+        val numberOfCats = retrievedCategories.size
+
+        assert(numberOfCats > 0)
+
+        val randomIndex = Random.nextInt(numberOfCats)
+        val expectedCategory = retrievedCategories[randomIndex]
```
Comment (edouardmichelin) [inline app/src/androidTest/java/com/studhub/app/CategoryRepositoryImplTest.kt:42]:
```kt
val expectedCategory = retrievedCategories.first { randomIndex == it.id }
```

**Review by edouardmichelin — CHANGES_REQUESTED** [review]

**Review by edouardmichelin — APPROVED** [review]
LGTM

**Review by Edouard-6 — APPROVED** [review]
LGTM

**Pull request merged**